src/qibo/validation/gate_set_tomography.py

```diff
@@ -0,0 +1,2089 @@
+                if not os.path.exists(GST_nogates_qibo_1qb):
+                    os.makedirs(GST_nogates_qibo_1qb)
+
+            elif class_qubit_gate is not None:
+                filedirectory = "GST_gates_qibo_1qb/"
+                if theta is None:
+                    fname = "%s gate (k=%.2d,j=%.2d) %s_basis.txt" % (
+                        class_qubit_gate,
+                        k,
+                        j,
+                        idx_basis[j],
+                    )
+                elif theta is not None:
+                    fname = "%s(%.12f) gate (k=%.2d,j=%.2d) %s_basis.txt" % (
+                        class_qubit_gate,
+                        theta,
+                        k,
+                        j,
+                        idx_basis[j],
+                    )
+                GST_gates_qibo_1qb = "GST_gates_qibo_1qb"
+                if not os.path.exists(GST_gates_qibo_1qb):
+                    os.makedirs(GST_gates_qibo_1qb)
+
+            # Save counts into a matrix. 0th column: word; 1st column: count value
+            # --------------------------------------------------------------------
+            fout = filedirectory + fname
+            fo = open(fout, "w")
+            for v1, v2 in counts.items():
+                fo.write(str(v1) + "  " + str(v2) + "\n")
+            fo.close()
+
+        # Find value for matrix_jk
+        # -------------------------
+
+        expectation_signs = np.matrix([[1, 1, 1, 1], [1, -1, -1, -1]])
+
+        matrix = np.zeros((2, 2))
+        for ii in range(0, 2):
+            word = bin(ii)[2:]
+            matrix[ii, 0] = word
+
+        for v1, v2 in counts.items():
+            # print(v1, v2)
+            row = int(v1, 2)
+            col = v2
+            matrix[row, 1] = col
+
+        sorted_matrix = sort_counts_1qb(matrix)
+        probs = sorted_matrix[:, 1] / np.sum(sorted_matrix[:, 1])
+        sorted_matrix = np.column_stack((sorted_matrix, probs))
+
+        k = k + 1
+        j = j + 1
+
+        if j == 1:
+            sorted_matrix = np.column_stack((sorted_matrix, expectation_signs[:, 0]))
+        elif j == 2:
+            sorted_matrix = np.column_stack((sorted_matrix, expectation_signs[:, 1]))
+        elif j == 3:
+            sorted_matrix = np.column_stack((sorted_matrix, expectation_signs[:, 1]))
+        elif j == 4:
+            sorted_matrix = np.column_stack((sorted_matrix, expectation_signs[:, 1]))
+        elif j == 5:
+            sorted_matrix = np.column_stack((sorted_matrix, expectation_signs[:, 2]))
+        elif j == 9:
+            sorted_matrix = np.column_stack((sorted_matrix, expectation_signs[:, 2]))
+        elif j == 13:
+            sorted_matrix = np.column_stack((sorted_matrix, expectation_signs[:, 2]))
+        else:
+            sorted_matrix = np.column_stack((sorted_matrix, expectation_signs[:, 3]))
+
+        temp = np.array(sorted_matrix[:, 2]) * np.array(sorted_matrix[:, 3])
```

Comment:
```suggestion
        temp = sorted_matrix[:, 2] * sorted_matrix[:, 3]
```
isn't sorted _matrix a ``np.ndarray`` already?

Reply:
This is the sorted_matrix after using ```np.column_stack``` to stack ```sorted_matrix``` and ```expectation_signs[:, ii]```:
![Screenshot 2023-11-30 at 11 13 04 AM](https://github.com/qiboteam/qibo/assets/46650770/ea6c153a-ab71-4dc4-94b2-0d0165f6016a)

The goal is to multiply the 2nd and 3rd column (zero based indexing) to get this result:
![Screenshot 2023-11-30 at 11 13 08 AM](https://github.com/qiboteam/qibo/assets/46650770/16878f0e-97cb-456c-9b02-cc3fdbcc030e)

I used ```temp = np.array(sorted_matrix[:, 2]) * np.array(sorted_matrix[:, 3])``` to get that result.

I've tried ```temp = sorted_matrix[:, 2] * sorted_matrix[:, 3]``` but an error pops up: ```ValueError: shapes (4,1) and (4,1) not aligned: 1 (dim 1) != 4 (dim 0)```.

Reply:
Yes because, ``sorted_matrix`` is probably an ``np.matrix`` which overloads the ``*`` operator with the matrix multiplication, thus, it's essentially the same as using @ with numpy arrays. This is the reason why I would use ``np.arrays`` instead of ``np.matrices`` in the first place.

Reply:
I've changed all the ```np.matrices``` to ```np.arrays``` and done extensive checks to ensure they're ok. :-)